File: jar-service/src/main/java/com/sixsq/slipstream/event/Event.java
package com.sixsq.slipstream.event;

import java.util.Date;
import java.util.HashMap;
import java.util.Map;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

public class Event {

	// TODO ???
	private static final String EVENT_URI = "http://sixsq.com/slipstream/1/Event";
	
	private static final String ISO_8601_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
	
	@SuppressWarnings("unused")
	private ACL acl;
	
	@SuppressWarnings("unused")
	private Date timestamp;
	
	@SuppressWarnings("unused")
	private String resourceURI;
	
	@SuppressWarnings("unused")
	private String resourceRef;
	
	@SuppressWarnings("unused")
	private String state;
	
	@SuppressWarnings("unused")
	private Severity severity;
	
	@SuppressWarnings("unused")
	private Map<String, Object> content;
	
	@SuppressWarnings("unused")
	private EventType type;
	
	public static enum Severity {
		critical, high, medium, low;
	}
	
	public static enum EventType {
		state, alarm;
	}
	
	public Event(ACL acl, Date timestamp, String resourceRef, String state, Severity severity, EventType type){	
		this.acl = acl;
		this.timestamp = timestamp;	
		this.resourceURI = EVENT_URI;		
		this.content = buildContent(resourceRef, state); 				
		this.severity = severity;
		this.type = type;
	}

	private Map<String, Object> buildContent(String resourceRef, String state) {
		Map<String, Object> result = new HashMap<String, Object>();	
		Map<String, String> resource = new HashMap<String, String>();
		resource.put("href", resourceRef);		
		result.put("resource", resource);		
		result.put("state", state);
		return result;
	}
	
	public String toJson(){

		Gson gson = new GsonBuilder()
		   .setDateFormat(ISO_8601_PATTERN)
		   .setPrettyPrinting()
		   .create();
		
		return gson.toJson(this);
	}
}

File: ssclj/jar/src/test/clojure/com/sixsq/slipstream/ssclj/resources/seeds/event.clj
(ns com.sixsq.slipstream.ssclj.resources.seeds.event
  (:require
    [clojure.data.json :as json]
    [peridot.core :refer :all]
    [clj-time.core :as time]
    [com.sixsq.slipstream.ssclj.resources.event :as e]
    [com.sixsq.slipstream.ssclj.app.params :as p]
    [com.sixsq.slipstream.ssclj.middleware.authn-info-header :as aih]
    [com.sixsq.slipstream.ssclj.resources.test-utils :as tu]
    [com.sixsq.slipstream.ssclj.db.impl :as db]
    [com.sixsq.slipstream.ssclj.resources.common.utils :as u]
    [com.sixsq.slipstream.ssclj.es.es-binding :as esb]))

(def base-uri (str p/service-context (u/de-camelcase e/resource-name)))

(defn- rnd-date-str
  []
  (-> 1000000 rand-int time/seconds time/from-now str))

(defn- event-template
  []
  {
   :acl       {
               :owner {
                       :type "USER" :principal :placeholder}
               :rules [{:type "USER" :principal :placeholder :right "ALL"}]}
   :timestamp (rnd-date-str)
   :content   {
               :resource {:href :placeholder}
               :state    "Started"}
   :type      "state"
   :severity  (rand-nth ["critical" "high" "medium" "low"])})

(defn events
  [nb-events username]
  (for [i (range nb-events)]
    (-> (event-template)
        (assoc-in [:acl :owner :principal] (name username))
        (assoc-in [:acl :rules 0 :principal] (name username))
        (assoc-in [:content :resource :href] (str "run/" i)))))


(defn insert-to-db
  [events username]
  (let [state (-> (session (tu/ring-app))
                  (content-type "application/json")
                  (header aih/authn-info-header username))]
    (doseq [event events]
      (request state base-uri
               :request-method :post
               :body (json/write-str event)))))

(defn seed!
  [nb-events username & {:keys [clean]}]
  (db/set-impl! (esb/get-instance))
  ;;(when clean)
    ;; TODO ES equivalent (kc/delete dbdb/resources))
  (-> nb-events
      (events username)
      (insert-to-db username)))
